implementation/src/main/java/io/smallrye/config/WithKeys.java

```diff
@@ -0,0 +1,31 @@
+package io.smallrye.config;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * Provides a list of map keys when populating {@link java.util.Map} types. The provided list will effectively
+ * substitute the lookup in {@link SmallRyeConfig#getPropertyNames()} for map keys. Each key must exist in the final
+ * configuration (relative in the mapping path segment), or the mapping will fail with a
+ * {@link java.util.NoSuchElementException}.
+ */
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.METHOD, ElementType.TYPE_USE })
```

Comment:
It should only be `TYPE_USE` IMO. The other cases where we have both are historical (IIRC) where it was wrongly `METHOD` but we actually intended it to be `TYPE_USE`. Having both means we have to check for it twice in very different ways, which is sometimes quite painful.

Reply:
Sure, this was to make it consistent with the other annotations.